### PR TITLE
Formatting of impl code using Jakarta EE formatting rules

### PR DESCRIPTION
**Pull request by arjantijms**
Signed-off-by: arjantijms <arjan.tijms@gmail.com>

### CODE REVIEW

File: impl/src/main/java/org/apache/jasper/runtime/TldScanner.java [hunk 1]
```diff
@@ -54,33 +54,21 @@
...
- * is maintained in this container.
- * Actually, that's what we'd like to do. However, because of the
- * way the classes TagLibraryInfo and TagInfo have been defined,
- * it is not currently possible to share an instance of TagLibraryInfo
- * across page invocations. A bug has been submitted to the spec lead.
- * In the mean time, all we do is save the 'location' where the
- * TLD associated with a taglib URI can be found.
+ * Tag Libraries can be defined globally in one of two ways: 1. Via <taglib> elements in web.xml: the uri and location
+ * of the tag-library are specified in the <taglib> element. 2. Via packaged jar files that contain .tld files within
+ * the META-INF directory, or some subdirectory of it. The taglib is 'global' if it has the <uri> element defined.
```
Comment (joakime) [inline impl/src/main/java/org/apache/jasper/runtime/TldScanner.java:61]: Numbered sections here got clobbered.

File: impl/src/main/java/org/apache/jasper/runtime/TldScanner.java [hunk 2]
```diff
@@ -201,41 +177,35 @@ public void onStartup(java.util.Set<java.lang.Class<?>> c,
...
+     * associated with any tag library 'exposed' in the web application.
      *
-     * This method may be called when the scanning is in one of states:
-     * 1. Called from jspc script, then a full tld scan is required.
-     * 2. The is the first call after servlet initialization, then system jars
-     *    that are knwon to have tlds but not listeners need to be scanned.
-     * 3. Sebsequent calls, no need to scans.
+     * This method may be called when the scanning is in one of states: 1. Called from jspc script, then a full tld scan is
+     * required. 2. The is the first call after servlet initialization, then system jars that are knwon to have tlds but not
+     * listeners need to be scanned. 3. Sebsequent calls, no need to scans.
```
Comment (joakime) [inline impl/src/main/java/org/apache/jasper/runtime/TldScanner.java:192]: Numbered sections got clobbered.

File: impl/src/main/java/org/apache/jasper/xmlparser/XMLEncodingDetector.java [hunk 3]
```diff
@@ -391,439 +363,403 @@ public boolean isExternal() {
...
+                        offset = 0;
+                        fCurrentEntity.position = newlines;
+                        if (load(newlines, false)) {
+                            break;
+                        }
+                    }
+                    /***
+                     * NEWLINE NORMALIZATION *** if (fCurrentEntity.ch[fCurrentEntity.position] == '\r' && external) {
+                     * fCurrentEntity.position++; offset++; } /
+                     **/
```
Comment (joakime) [inline impl/src/main/java/org/apache/jasper/xmlparser/XMLEncodingDetector.java]: Commented code?

File: impl/src/main/java/org/apache/jasper/xmlparser/XMLEncodingDetector.java [hunk 4]
```diff
@@ -886,115 +820,108 @@ public boolean skipChar(int c) throws IOException {
...
+                        // REVISIT: Does this need to be updated to fix the
+                        // #x0D ^#x0A newline normalization problem? -Ac
+                        if (fCurrentEntity.ch[++fCurrentEntity.position] != '\n') {
+                            fCurrentEntity.position--;
+                        }
+                    }
+                    /***
+                     * NEWLINE NORMALIZATION *** else { if (fCurrentEntity.ch[fCurrentEntity.position + 1] == '\r' && external) {
+                     * fCurrentEntity.position++; } } /
+                     **/
```
Comment (joakime) [inline impl/src/main/java/org/apache/jasper/xmlparser/XMLEncodingDetector.java]: Commented code?

File: impl/src/main/java/org/apache/jasper/xmlparser/ParserUtils.java [hunk 5]
```diff
@@ -88,44 +87,26 @@
...
-    static final String[] CACHED_SCHEMA_RESOURCE_PATHS =
-            (String[])DEFAULT_SCHEMA_RESOURCE_PATHS; 
-    // END PWC 6386258
+    private static final String[] DEFAULT_DTD_RESOURCE_PATHS = { Constants.TAGLIB_DTD_RESOURCE_PATH_11, Constants.TAGLIB_DTD_RESOURCE_PATH_12,
+            Constants.WEBAPP_DTD_RESOURCE_PATH_22, Constants.WEBAPP_DTD_RESOURCE_PATH_23, };
+
+    static final String[] CACHED_DTD_RESOURCE_PATHS = (String[]) DEFAULT_DTD_RESOURCE_PATHS;
 
+    private static final String[] DEFAULT_SCHEMA_RESOURCE_PATHS = { Constants.TAGLIB_SCHEMA_RESOURCE_PATH_20, Constants.TAGLIB_SCHEMA_RESOURCE_PATH_21,
+            Constants.WEBAPP_SCHEMA_RESOURCE_PATH_24, Constants.WEBAPP_SCHEMA_RESOURCE_PATH_25, };
```
Comment (joakime) [inline impl/src/main/java/org/apache/jasper/xmlparser/ParserUtils.java:106]: The array formatting issue again.

File: impl/src/main/java/org/apache/jasper/tagplugins/jstl/ForEach.java [hunk 6]
```diff
@@ -144,200 +142,114 @@ private void doCollection(TagPluginContext ctxt) {
...
-	    "      return e.nextElement();}\n" +
-	    "    public void remove() {}\n" +
-	    "  });\n" +
-	    "}"
-	);
+        // Object[]
+        ctxt.generateDeclaration("ObjectArrayIterator",
+                "private Iterator toIterator(final Object[] a){\n" + "  return (new Iterator() {\n" + "    int index=0;\n" + "    public boolean hasNext() {\n"
+                        + "      return index < a.length;}\n" + "    public Object next() {\n" + "      return a[index++];}\n" + "    public void remove() {}\n"
+                        + "  });\n" + "}");
```
Comment (joakime) [inline impl/src/main/java/org/apache/jasper/tagplugins/jstl/ForEach.java:198]: This array reformatting made it more difficult to read this section.

File: impl/src/main/java/org/apache/jasper/tagplugins/jstl/ForEach.java [hunk 7]
```diff
@@ -144,200 +142,114 @@ private void doCollection(TagPluginContext ctxt) {
...
+
+        // boolean[]
+        ctxt.generateDeclaration("booleanArrayIterator",
+                "private Iterator toIterator(final boolean[] a){\n" + "  return (new Iterator() {\n" + "    int index=0;\n" + "    public boolean hasNext() {\n"
+                        + "      return index < a.length;}\n" + "    public Object next() {\n" + "      return new Boolean(a[index++]);}\n"
+                        + "    public void remove() {}\n" + "  });\n" + "}");
```
Comment (joakime) [inline impl/src/main/java/org/apache/jasper/tagplugins/jstl/ForEach.java:204]: Ditto, array reformatting.

File: impl/src/main/java/org/apache/jasper/tagplugins/jstl/ForEach.java [hunk 8]
```diff
@@ -144,200 +142,114 @@ private void doCollection(TagPluginContext ctxt) {
...
+
+        // byte[]
+        ctxt.generateDeclaration("byteArrayIterator",
+                "private Iterator toIterator(final byte[] a){\n" + "  return (new Iterator() {\n" + "    int index=0;\n" + "    public boolean hasNext() {\n"
+                        + "      return index < a.length;}\n" + "    public Object next() {\n" + "      return new Byte(a[index++]);}\n"
+                        + "    public void remove() {}\n" + "  });\n" + "}");
```
Comment (joakime) [inline impl/src/main/java/org/apache/jasper/tagplugins/jstl/ForEach.java:210]: Ditto, array reformatting.

File: impl/src/main/java/org/apache/jasper/tagplugins/jstl/ForEach.java [hunk 9]
```diff
@@ -144,200 +142,114 @@ private void doCollection(TagPluginContext ctxt) {
...
+
+        // char[]
+        ctxt.generateDeclaration("charArrayIterator",
+                "private Iterator toIterator(final char[] a){\n" + "  return (new Iterator() {\n" + "    int index=0;\n" + "    public boolean hasNext() {\n"
+                        + "      return index < a.length;}\n" + "    public Object next() {\n" + "      return new Character(a[index++]);}\n"
+                        + "    public void remove() {}\n" + "  });\n" + "}");
```
Comment (joakime) [inline impl/src/main/java/org/apache/jasper/tagplugins/jstl/ForEach.java:216]: Ditto, array reformatting.

File: impl/src/main/java/org/apache/jasper/tagplugins/jstl/ForEach.java [hunk 10]
```diff
@@ -144,200 +142,114 @@ private void doCollection(TagPluginContext ctxt) {
...
+
+        // short[]
+        ctxt.generateDeclaration("shortArrayIterator",
+                "private Iterator toIterator(final short[] a){\n" + "  return (new Iterator() {\n" + "    int index=0;\n" + "    public boolean hasNext() {\n"
+                        + "      return index < a.length;}\n" + "    public Object next() {\n" + "      return new Short(a[index++]);}\n"
+                        + "    public void remove() {}\n" + "  });\n" + "}");
```
Comment (joakime) [inline impl/src/main/java/org/apache/jasper/tagplugins/jstl/ForEach.java:222]: Ditto, array reformatting.

File: impl/src/main/java/org/apache/jasper/tagplugins/jstl/ForEach.java [hunk 11]
```diff
@@ -144,200 +142,114 @@ private void doCollection(TagPluginContext ctxt) {
...
+
+        // int[]
+        ctxt.generateDeclaration("intArrayIterator",
+                "private Iterator toIterator(final int[] a){\n" + "  return (new Iterator() {\n" + "    int index=0;\n" + "    public boolean hasNext() {\n"
+                        + "      return index < a.length;}\n" + "    public Object next() {\n" + "      return new Integer(a[index++]);}\n"
+                        + "    public void remove() {}\n" + "  });\n" + "}");
```
Comment (joakime) [inline impl/src/main/java/org/apache/jasper/tagplugins/jstl/ForEach.java:228]: Ditto, array reformatting.

File: impl/src/main/java/org/apache/jasper/tagplugins/jstl/ForEach.java [hunk 12]
```diff
@@ -144,200 +142,114 @@ private void doCollection(TagPluginContext ctxt) {
...
+
+        // long[]
+        ctxt.generateDeclaration("longArrayIterator",
+                "private Iterator toIterator(final long[] a){\n" + "  return (new Iterator() {\n" + "    int index=0;\n" + "    public boolean hasNext() {\n"
+                        + "      return index < a.length;}\n" + "    public Object next() {\n" + "      return new Long(a[index++]);}\n"
+                        + "    public void remove() {}\n" + "  });\n" + "}");
```
Comment (joakime) [inline impl/src/main/java/org/apache/jasper/tagplugins/jstl/ForEach.java:234]: Ditto, array reformatting.

File: impl/src/main/java/org/apache/jasper/tagplugins/jstl/ForEach.java [hunk 13]
```diff
@@ -144,200 +142,114 @@ private void doCollection(TagPluginContext ctxt) {
...
+
+        // float[]
+        ctxt.generateDeclaration("floatArrayIterator",
+                "private Iterator toIterator(final float[] a){\n" + "  return (new Iterator() {\n" + "    int index=0;\n" + "    public boolean hasNext() {\n"
+                        + "      return index < a.length;}\n" + "    public Object next() {\n" + "      return new Float(a[index++]);}\n"
+                        + "    public void remove() {}\n" + "  });\n" + "}");
```
Comment (joakime) [inline impl/src/main/java/org/apache/jasper/tagplugins/jstl/ForEach.java:240]: Ditto, array reformatting.

File: impl/src/main/java/org/apache/jasper/tagplugins/jstl/ForEach.java [hunk 14]
```diff
@@ -144,200 +142,114 @@ private void doCollection(TagPluginContext ctxt) {
...
+
+        // double[]
+        ctxt.generateDeclaration("doubleArrayIterator",
+                "private Iterator toIterator(final double[] a){\n" + "  return (new Iterator() {\n" + "    int index=0;\n" + "    public boolean hasNext() {\n"
+                        + "      return index < a.length;}\n" + "    public Object next() {\n" + "      return new Double(a[index++]);}\n"
+                        + "    public void remove() {}\n" + "  });\n" + "}");
```
Comment (joakime) [inline impl/src/main/java/org/apache/jasper/tagplugins/jstl/ForEach.java:246]: Ditto, array reformatting.

File: impl/src/main/java/org/apache/jasper/tagplugins/jstl/ForEach.java [hunk 15]
```diff
@@ -144,200 +142,114 @@ private void doCollection(TagPluginContext ctxt) {
...
+
+        // Enumeration
+        ctxt.generateDeclaration("enumIterator",
+                "private Iterator toIterator(final Enumeration e){\n" + "  return (new Iterator() {\n" + "    public boolean hasNext() {\n"
+                        + "      return e.hasMoreElements();}\n" + "    public Object next() {\n" + "      return e.nextElement();}\n"
+                        + "    public void remove() {}\n" + "  });\n" + "}");
```
Comment (joakime) [inline impl/src/main/java/org/apache/jasper/tagplugins/jstl/ForEach.java:252]: Ditto, array reformatting.

File: impl/src/main/java/org/apache/jasper/servlet/JasperLoader.java [hunk 16]
```diff
@@ -166,23 +152,20 @@ public Class findClass(String className) throws ClassNotFoundException {
         }
 
         // Preprocess the loaded byte code
-/* XXX
-        if (PreprocessorUtil.isPreprocessorEnabled()) {
-            cdata = PreprocessorUtil.processClass(path, cdata);
-        }
-*/
+        /*
+         * XXX if (PreprocessorUtil.isPreprocessorEnabled()) { cdata = PreprocessorUtil.processClass(path, cdata); }
```
Comment (joakime) [inline impl/src/main/java/org/apache/jasper/servlet/JasperLoader.java:156]: Commented code?

File: impl/src/main/java/org/apache/jasper/servlet/JasperLoader.java [hunk 17]
```diff
@@ -151,12 +139,10 @@ public Class findClass(String className) throws ClassNotFoundException {
...
             // If the bytecode preprocessor is not enabled, use super.findClass
-   	    // as usual.
-/* XXX
-            if (!PreprocessorUtil.isPreprocessorEnabled()) { 
-                return super.findClass(className);
-            }
-*/
+            // as usual.
+            /*
+             * XXX if (!PreprocessorUtil.isPreprocessorEnabled()) { return super.findClass(className); }
```
Comment (joakime) [inline impl/src/main/java/org/apache/jasper/servlet/JasperLoader.java:144]: Commented code?

**Comment by arjantijms** [comment]
> Lots of commented code sections. I say remove them, as they already exist in the source control, there's no reason to keep them around and being merged into comments (sometimes even into javadoc visible comments)

Definitely agree with that, though I somewhat planned to do that in a later PR. I can do them here too though.

**Comment by markt-asf** [comment]
Maybe do the obvious clean-up like removing commented out code first and then apply the formatting rules?
If some of the rules don't make sense for our code we don;t have to follow them but we'll need to document what we agree. Maybe in CONTRIBUTING ?

**Pull request closed without merging**